theano/gof/opt.py

```diff
@@ -2669,7 +2674,9 @@ def check_stack_trace(f_or_fgraph, ops_to_check='last', bug_print='raise'):
     # if ops_to_check is a list/tuple of ops
     elif isinstance(ops_to_check, (tuple, list)):
         apply_nodes_to_check = [node for node in fgraph.apply_nodes
-                                if node.op in ops_to_check]
+                                if node.op in ops_to_check or
+                                isinstance(node.op, ops_to_check) or
+                                isinstance(node.op.scalar_op, ops_to_check)]
```

Comment:
Wait, why does it make sense to check `node.op.scalar_op`?
Also, not all `Op` instances have a `scalar_op` attribute.